Mathlib/Topology/Subbasis.lean

```diff
@@ -0,0 +1,40 @@
+/-
+Copyright (c) 2025 Fangming Li. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Fangming Li
+-/
+import Mathlib.Topology.Compactness.Compact
+
+/-!
+# Results on subbases of topological spaces
+
+In this file, we prove results when some topology is given a subbasis.
+
+## Main results
+
+* `alexander_subbasis hTS`. If `X` has a topology `T` and `hTS : T = generateFrom S`, then `X` is
+  compact if any open cover of `X` with elements in `S` has a finite subcover.
+-/
+
+open TopologicalSpace
+
+variable {X : Type*} [T : TopologicalSpace X] {S : Set (Set X)}
+
+/--
+The Alexander Subbasis Theorem. If `X` is a topological space with a subbasis `S`, then `X` is
+compact if any open cover of `X` with elements in `S` has a finite subcover.
+-/
+theorem alexander_subbasis (hTS : T = generateFrom S) :
+    (∀ P ⊆ S, ⋃₀ P = .univ → ∃ Q ⊆ P, Q.Finite ∧ ⋃₀ Q = .univ) → CompactSpace X := by
```

Comment:
```suggestion
theorem alexander_subbasis (hTS : T = generateFrom S)
    (h : ∀ P ⊆ S, ⋃₀ P = .univ → ∃ Q ⊆ P, Q.Finite ∧ ⋃₀ Q = .univ) : CompactSpace X := by
```
And we might want a different name such as `compactSpace_generateFrom`.
Another thing you could do is drop the topology and do
```suggestion
theorem alexander_subbasis
    (h : ∀ P ⊆ S, ⋃₀ P = .univ → ∃ Q ⊆ P, Q.Finite ∧ ⋃₀ Q = .univ) : @CompactSpace X (generateFrom S) := by
```